src/util/output_channels.ts

```diff
@@ -45,6 +50,57 @@ export function openNamedOutputChannel(name?: string): vscode.OutputChannel | un
   return channel;
 }
 
+async function credHelper(startProcess: ChildProcess): Promise<void> {
+  const resourceGroups: vscode.QuickPickItem[] = [
+    'docker-credential-desktop',
```

Comment:
Helpers shouldn't be hard-coded.

```js
const path = require('node:path');
const { readdir } = require('node:fs/promises');

const getHelpers = async () => {
    const isCredHelper = f => f.startsWith("docker-credential")
    let helpers = []
    for (const dir of process.env.PATH.split(path.delimiter)) {
        try {
            const files = await readdir(dir)
            helpers = helpers.concat(files.filter(isCredHelper))
        } catch (e) {
            console.log(e)
        }
    }
    return new Set(helpers)
}
```

Reply:
done

Reply:
Given the lint errors we might use:

```ts
export function getHelpers(): Promise<Array<string>> {
  const isCredHelper = (f: string) => f.startsWith('docker-credential');

  return process.env.PATH.split(path.delimiter)
    .map(async (dir) => {
      try {
        return (await readdir(dir)).filter(isCredHelper);
      } catch (e) {
        // maybe log error via some other facility than `console`
        return [];
      }
    })
    .reduce(async (a, b) => (await a).concat(await b));
}
```

Reply:
or maybe we can disregard the error all together,
the error happens only if the directory on $PATH doesn't exist or is inaccessible, should happen rarely.